Files/ViewModels/Bundles/BundleContainerViewModel.cs

```diff
@@ -0,0 +1,329 @@
+﻿using Files.SettingsInterfaces;
+using Microsoft.Toolkit.Mvvm.ComponentModel;
+using Microsoft.Toolkit.Mvvm.Input;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Windows.Input;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.Storage;
+using Windows.System;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Input;
+
+namespace Files.ViewModels.Bundles
+{
+	/// <summary>
+	/// Bundle's contents view model
+	/// </summary>
+	public class BundleContainerViewModel : ObservableObject, IDisposable
+	{
+		#region Singleton
+
+		private IJsonSettings JsonSettings => associatedInstance?.InstanceViewModel.JsonSettings;
+
+		#endregion
+
+		#region Private Members
+
+		private IShellPage associatedInstance;
+
+		#endregion
+
+		#region Events
+
+		public Action<BundleContainerViewModel> NotifyItemRemoved { get; set; }
+
+		public Action<BundleContainerViewModel> NotifyItemRenamed { get; set; }
+
+		#endregion
+
+		#region Public Properties
+
+		/// <summary>
+		/// A list of Bundle's contents
+		/// </summary>
+		public ObservableCollection<BundleItemViewModel> Contents { get; private set; } = new ObservableCollection<BundleItemViewModel>();
+
+		private string pBundleName = "DefaultBundle";
+		public string BundleName
+		{
+			get => pBundleName;
+			set => SetProperty(ref pBundleName, value);
+		}
+
+		private Visibility pNoBundleContentsTextVisibility;
+		public Visibility NoBundleContentsTextVisibility
+		{
+			get => pNoBundleContentsTextVisibility;
+			set => SetProperty(ref pNoBundleContentsTextVisibility, value);
+		}
+
+		private string pBundleRenameText = string.Empty;
+		public string BundleRenameText
+		{
+			get => pBundleRenameText;
+			set => SetProperty(ref pBundleRenameText, value);
+		}
+
+		private Visibility _BundleRenameVisibility = Visibility.Collapsed;
```

Comment:
Check code style, please.

Reply:
sorry, it slid past unnoticed. I'll fix it